src/shm_main.c

```diff
@@ -43,6 +43,22 @@ struct shm_item {
     char *name;
 };
 
+/**
+ * @brief Item holding information about active connections owned by this process.
+ */
+typedef struct _conn_list_entry conn_list_entry;
+struct _conn_list_entry {
+    conn_list_entry *_next;
+    sr_cid_t cid;
+    int cid_fd;
+};
+
+/**
+ * @brief Linked list of all active connections in this process.
+ */
+static conn_list_entry *conn_head = NULL;
+static pthread_mutex_t *conn_list_lock = NULL;
```

Comment:
The way I understood it, the whole purpose of this linked list is an optimization so that to test the aliveness of a connection from the same process it is not necessary to call `open`. Is that correct, is it the only purpose?

Reply:
The reason this list is maintained is that I need to track which connections are owned by the current process.  The function sr_shmmain_check_lock makes use of this list to return "alive" for any connection owned by the current process.  For each connection in the current process we open an FD and hold the lock which indicates that our connection is "alive".  Closing **any** FD to that same file will immediately clear the lock (normal behavior of advisory locks).  This means that in sr_shmmain_check_conn_lock we cannot simply open a new FD (based on the CID) to the file and check the lock.  Instead I maintain this list of CID owned by the current process and use that to determine "aliveness" -- thus avoiding the need to opening and closing an FD and inadvertently releasing the lock.  
This list is also a convenient place to store the FD used in opening/holding the lock on the file.  The sr_shmmain_deallocate_conn uses this list to retrieve the FD, which it then closes, releasing the lock and freeing the FD.

Reply:
Okay, makes sense. So, please put the two attributes into a (anonymous) structure to keep it clean. Also, please try to summarize what you just wrote somewhere in the code. Maybe the docs of `sr_shmmain_check_conn_lock` would be the place. It does not matter much, as long as it is somewhere.

Reply:
Done.